discovery/gossiper.go

```diff
@@ -3423,3 +3397,42 @@ func (d *AuthenticatedGossiper) saveChanUpdateToDisk(upd *lnwire.ChannelUpdate,
 
 	return err
 }
+
+// prematureMsgKey defines the key to be used when saving items to premature
+// message cache.
+type prematureMsgKey interface {
+	// We use either a shortChanID for ChannelUpdate or node pubKey for
+	// NodeAnnouncement.
+	uint64 | [33]byte
+}
+
+// savePrematureMsg puts a message in the premature message cache.
+func savePrematureMsg[T prematureMsgKey](d *AuthenticatedGossiper,
+	key T, nMsg *networkMsg) {
+
+	pMsg := &processedNetworkMsg{msg: nMsg}
+
+	// The shortChanID may be an alias, but it is fine to use here since we
+	// don't have an edge in the graph and if the peer is not buggy, we
+	// should be able to use it once the gossiper receives the local
+	// announcement.
+	earlyMsgs, err := d.prematureMessages.Get(key)
+	switch {
+	// Nothing in the cache yet, we can just directly insert this element.
+	case errors.Is(err, cache.ErrElementNotFound):
+		_, _ = d.prematureMessages.Put(
+			key, &cachedNetworkMsg{
+				msgs: []*processedNetworkMsg{pMsg},
+			})
+
+	// There's already something in the cache, so we'll combine the set of
+	// messages into a single value.
+	default:
+		msgs := earlyMsgs.(*cachedNetworkMsg).msgs
+		msgs = append(msgs, pMsg)
+		_, _ = d.prematureMessages.Put(
+			key, &cachedNetworkMsg{
+				msgs: msgs,
+			})
```

Comment:
Closing the paren on the new line? I think also maybe the params can be on the first line of `Put`?

Reply:
Changed!